graffiti/storage/elasticsearch/client.go

```diff
@@ -532,7 +532,16 @@ func (c *Client) Search(query elastic.Query, opts filters.SearchQuery, indices .
 		})
 	}
 
-	return searchQuery.Do(context.Background())
+	res, err := searchQuery.Do(context.Background())
+	// Detect partial failures (only some shards failing)
+	if err == nil && res.Shards.Failed != 0 {
+		errors := []string{}
+		for _, f := range res.Shards.Failures {
+			errors = append(errors, fmt.Sprintf("reason='%v'", f.Reason))
```

Comment:
What about using github.com/hashicorp/go-multierror for this ?

Reply:
I didn't know that lib. Fixed!
Thanks